### PR TITLE
Tokenise query using Text tokeniser, search according to property datatype

### PR DESCRIPTION
**Pull request by donomii**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [x] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [x] Performance tests have been run or not necessary.

This fixes tokenising on the input query for bm25

### CODE REVIEW

File: adapters/repos/db/bm25f_test.go [hunk 1]
```diff
@@ -204,7 +205,17 @@ func TestBM25FJourney(t *testing.T) {
...
+	})
+
+	fmt.Println("Search with non alphanums")
+	// Check search with no properties (should include all properties)
+	kwr = &searchparams.KeywordRanking{Type: "bm25", Properties: []string{}, Query: "*&^$@#$%^&*()(offtopic!!!!"}
+	res, err = idx.objectSearch(context.TODO(), 1000, nil, kwr, nil, addit)
+	require.Nil(t, err)
+
+	t.Run("bm25f non alphanums", func(t *testing.T) {
+		fmt.Printf("Results: %+v", res[0])
```
Comment (antas-marcin) [inline adapters/repos/db/bm25f_test.go:217]: do we need this `fmt.Printf` statement?

File: adapters/repos/db/bm25f_test.go [hunk 2]
```diff
@@ -204,7 +205,17 @@ func TestBM25FJourney(t *testing.T) {
 		require.Equal(t, uint64(4), res[1].DocID())
 		require.Equal(t, uint64(5), res[2].DocID())
 		require.Equal(t, uint64(6), res[3].DocID())
-		require.Equal(t, uint64(2), res[4].DocID())
+	})
+
+	fmt.Println("Search with non alphanums")
```
Comment (antas-marcin) [inline adapters/repos/db/bm25f_test.go:210]: do we need this `fmt.Println` statement?

**Review by antas-marcin — APPROVED** [review]

File: adapters/repos/db/bm25f_test.go (hunk 1 above)
Comment (donomii) [reply adapters/repos/db/bm25f_test.go:217]: removed

File: adapters/repos/db/bm25f_test.go (hunk 2 above)
Comment (donomii) [reply adapters/repos/db/bm25f_test.go:210]: This one, yes, it helps understand the output when a test fails.  It doesn't get printed unless the test fails

**Review by etiennedi — APPROVED** [review]
Looks good. I can't tell how well we're exhausting edge cases in the test(s). This PR seems to add just one more case, but there are already existing cases, so it could easily be that we're covering everything already.
